dashboard/test/controllers/featured_projects_controller_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class FeaturedProjectsControllerTest < ActionController::TestCase
+  setup do
+    @project_validator = create :project_validator
+    @featured_project = create :featured_project
+    @teacher = create :teacher
+  end
+
+  test 'project validators can feature projects' do
+    sign_in @project_validator
+    @controller.stubs(:storage_decrypt_channel_id).returns([123, 456])
+    put :feature, params: {project_id: 456}
+    assert_response :success
```

Comment:
This looks great. 

It's probably better / clearer to have the project_id and channel_id be different values.
We can make the test even stronger also by using expectations instead of stubs, so we validate `storage_decrypt_channel_id` was called and with the right arguments:

s/stubs/expects, and add `.with`
```ruby
@controller.expects(:storage_decrypt_channel_id).with('my project id').returns(['my storage id', 'my channel id'])
```

We should also validate that it did the thing it was supposed to do and featured a project.
You can wrap the `put` line in `assert_creates(FeaturedProject) do` to make sure it creates something, and then optionally assert properties on `FeaturedProject.last` at the end. This could be a separate test case to separate the auth from functionality if you want.

Reply:
Do we want a small test for idempotence too? What if feature is called on an already featured project?